pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -353,6 +354,20 @@ type WorkflowSpec struct {
 
 	// TemplateDefaults holds default template values that will apply to all templates in the Workflow, unless overridden on the template-level
 	TemplateDefaults *Template `json:"templateDefaults,omitempty" protobuf:"bytes,39,opt,name=templateDefaults"`
+
+	// ExitTemplate hold the template and arguments which is invoked at the end of the
+	// workflow, irrespective of the success, failure, or error status of workflow
+	ExitTemplate *ExitTemplate `json:"exitTemplate,omitempty" protobuf:"bytes,40,opt,name=exitTemplate"`
```

Comment:
@jessesuen @sarabala1979 I think this name will be confusing.

ExitTemplate is another overload of "template". Instead, what about this?

```yaml
spec:
  hooks:
    exit: 
      arguments: ...
      template: template-name
      # OR
      inline:  // wfv1.Template
```

Reply:
I think `ExitTemplate` makes easy understanding and directly conveys the meaning to the user.

Reply:
@jessesuen Please comment on above one

Reply:
I agree hook is probably a better term and if we do eventually want inline specs we will need @alexec 's proposed structure.

Reply:
```yaml
spec:
  hooks:
    exit: # dfv1.LifecycleHook
      arguments: ...
      template: template-name
      # OR
      inline: ... # wfv1.Template
```